src/Provider/AbstractProvider.php

```diff
@@ -202,12 +202,36 @@ public function getAccessToken($grant = 'authorization_code', $params = [])
             // @codeCoverageIgnoreEnd
         }
 
+        $result = $this->prepareResponse($response);
+
+        if (isset($result['error']) && ! empty($result['error'])) {
+            // @codeCoverageIgnoreStart
+            throw new IDPException($result);
+            // @codeCoverageIgnoreEnd
+        }
+
+        $result = $this->prepareAccessTokenResult($result);
+
+        return $grant->handleResponse($result);
+    }
+
+    /**
+     * Prepare the response, parsing according to configuration and returning
+     * the response as an array.
+     *
+     * @param  string $response
+     * @return array
+     */
+    protected function prepareResponse($response)
```

Comment:
Moved the response parsing to a separate method so that `fetchProviderData` could make use of it.